test/built-ins/TypedArray/prototype/findIndex/predicate-call-this-non-strict.js

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-%typedarray%.prototype.findindex
+es6id: 22.2.3.11
+description: >
+  Verify predicate this on non-strict mode
+info: >
+  22.2.3.11 %TypedArray%.prototype.findIndex ( predicate [ , thisArg ] )
+
+  %TypedArray%.prototype.findIndex is a distinct function that implements the
+  same algorithm as Array.prototype.findIndex as defined in 22.1.3.9 except that
+  the this object's [[ArrayLength]] internal slot is accessed in place of
+  performing a [[Get]] of "length".
+
+  ...
+
+  22.1.3.9 Array.prototype.findIndex ( predicate[ , thisArg ] )
+
+  ...
+  4. If thisArg was supplied, let T be thisArg; else let T be undefined.
+  5. Let k be 0.
+  6. Repeat, while k < len
+    ...
+    c. Let testResult be ToBoolean(? Call(predicate, T, « kValue, k, O »)).
+  ...
+flags: [noStrict]
+includes: [testTypedArray.js]
+---*/
+
+var T = this;
+
+testWithTypedArrayConstructors(function(TA) {
+  var sample = new TA(1);
+  var result;
+
+  sample.findIndex(function() {
+    result = this;
+  });
+
+  assert.sameValue(result, T, "without thisArg, predicate this is the global");
+
+  var o = {};
+  sample.findIndex(function() {
+    result = this;
+  }, o);
```

Comment:
+1. I could tell the non existent argument is handled as undefined, but it's precise adding a specific test for it.